include/xtensor/xiterable.hpp

```diff
@@ -264,8 +264,405 @@ namespace xt
         derived_type& derived_cast();
     };
 
+    template <class T>
+    struct xcontainer_inner_types;
+
+    template <class D>
+    class xcontiguous_iterable : public xiterable<D>
```

Comment:
it is not necessarily contiguous, but rather "container-based."

Reply:
container-based might imply inheriting form `xcontainer` while other kind of expressions might be contiguously iterable (without being a container).